lsp-mode.el

```diff
@@ -3961,9 +3961,10 @@ LSP server result."
                                                 (forward-line 0)
                                                 (insert offset)))
                                  indent-line-function)))
-    (yas-expand-snippet
-     (lsp--to-yasnippet-snippet snippet)
-     start end expand-env)))
+    (when (featurep 'yasnippet)
```

Comment:
here it is too late. If yasnippet is not present the completion candidate will stay with placeholders. I think that the bug here is that rust-analyzer supports only running with snippet support = true.

Reply:
> here it is too late.

Any suggestion for the fix? I can try implement it.

> I think that the bug here is that rust-analyzer supports only running with snippet support = true.

Not sure what you mean by this. Can you elaborate a bit more? Thank you!

Reply:
Check this section - https://github.com/yyoncho/lsp-mode/blob/rust-analyzer-plists/lsp-mode.el#L3263

If we don't have yasnippet, snippetSupport should be false and the server should not send snippets. So the bug is either in this section or in the server implementation.

Reply:
Also, wouldn't that break if yasnippet were lazy-loaded, and just not loaded yet? Something like fbound 'yas-minor-mode should be better.

Reply:
Yasnippet is required when loading `lsp-mode` with no error option. So unless Yasnippet is not install, there should be no issue here. We purposely to load Yasnippet before hand because it's slow and having it load here should cause significant lag when user types.

Reply:
> If we don't have yasnippet, snippetSupport should be false and the server should not send snippets. So the bug is either in this section or in the server implementation.

Ahh... I see what you mean here. So you prefer PR that prevents sending snippet request?

If this is the case, what could be wrong here? I wonder why this is happening? How come it leads to this operation. 😕 

> Yasnippet is required when loading lsp-mode with no error option. So unless Yasnippet is not install, there should be no issue here.

Got it! I have now much understand the code base. ;)

Reply:
I don't see a snippetSupport for code actions, so how would not sending snippetSupport for it fix #2281?

Reply:
> I don't see a snippetSupport for code actions, so how would not sending snippetSupport for it fix #2281?

Sending snippet or not is depended on server side. The client side (`lsp-mode`) will tell the server that it cannot support snippet if `yas-snippet` is not presented, now it up to the server side to uphold the LSP spec and do not send anything that contains snippet back